### PR TITLE
Added more generic loops so that adding newer version (v1.30) easier across all tests

### PR DESCRIPTION
**Pull request by deepakm-ntnx**
**What this PR does / why we need it**:
Added more generic loops so that adding newer version (v1.30) easier across all tests

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**How Has This Been Tested?**:
<pre>
make list-e2e LABEL_FILTERS=Kube130

RUN_VALIDATION_TESTS_ONLY=true make test-e2e-calico LABEL_FILTERS="Kube130 && cluster-topology-basic"

RUN_VALIDATION_TESTS_ONLY=true make test-e2e-calico LABEL_FILTERS="Kube130 && cluster-topology-conformance"

RUN_VALIDATION_TESTS_ONLY=true make test-e2e-calico LABEL_FILTERS="Kube130 && cluster-topology-scale-in-cp"
RUN_VALIDATION_TESTS_ONLY=true make test-e2e-calico LABEL_FILTERS="Kube130 && cluster-topology-scale-in-worker"
RUN_VALIDATION_TESTS_ONLY=true make test-e2e-calico LABEL_FILTERS="Kube130 && cluster-topology-scale-in-cp-worker"

RUN_VALIDATION_TESTS_ONLY=true make test-e2e-calico LABEL_FILTERS="Kube130 && cluster-topology-scale-out-cp"
RUN_VALIDATION_TESTS_ONLY=true make test-e2e-calico LABEL_FILTERS="Kube130 && cluster-topology-scale-out-worker"
RUN_VALIDATION_TESTS_ONLY=true make test-e2e-calico LABEL_FILTERS="Kube130 && cluster-topology-scale-out-cp-worker"

RUN_VALIDATION_TESTS_ONLY=true make test-e2e-calico LABEL_FILTERS="Kube130 && cluster-topology-scale-down-memory-size"
RUN_VALIDATION_TESTS_ONLY=true make test-e2e-calico LABEL_FILTERS="Kube130 && cluster-topology-scale-down-vcpusockets"
RUN_VALIDATION_TESTS_ONLY=true make test-e2e-calico LABEL_FILTERS="Kube130 && cluster-topology-scale-down-vcpupersocket"

RUN_VALIDATION_TESTS_ONLY=true make test-e2e-calico LABEL_FILTERS="Kube130 && cluster-topology-scale-up-memory-size"
RUN_VALIDATION_TESTS_ONLY=true make test-e2e-calico LABEL_FILTERS="Kube130 && cluster-topology-scale-up-vcpusockets"
RUN_VALIDATION_TESTS_ONLY=true make test-e2e-calico LABEL_FILTERS="Kube130 && cluster-topology-scale-up-vcpupersocket"

RUN_VALIDATION_TESTS_ONLY=true make test-e2e-calico LABEL_FILTERS="Kube130 && cluster-topology-upgrade"

</pre>

<pre>
RUN_VALIDATION_TESTS_ONLY=true make test-e2e-calico LABEL_FILTERS="Kube130 && cluster-topology-basic"
</pre>
<pre>
When creating a cluster with topology with simple workflow Create a cluster with topology with Kube130 [clusterclass, only-for-validation, cluster-topology-basic, Kube130]
/Users/deepak.muley/go/src/github.com/nutanix-cloud-native/cluster-api-provider-nutanix/test/e2e/cluster_topology_basic_test.go:98
  STEP: Creating a namespace for hosting the "cluster-with-topology-simple-workflow" test spec @ 05/01/24 13:17:47.367
  INFO: Creating namespace cluster-with-topology-simple-workflow-43cwdq
  INFO: Creating event watcher for namespace "cluster-with-topology-simple-workflow-43cwdq"
  STEP: Creating a workload cluster with topology @ 05/01/24 13:17:47.375
  INFO: Creating the workload cluster with name "cluster-with-topology-simple-workflow-5q05a5" using the "topology" template (Kubernetes v1.30.0, 1 control-plane machines, 1 worker machines)
  INFO: Getting the cluster template yaml
  INFO: clusterctl config cluster cluster-with-topology-simple-workflow-5q05a5 --infrastructure (default) --kubernetes-version v1.30.0 --control-plane-machine-count 1 --worker-machine-count 1 --flavor topology
  INFO: Creating the workload cluster with name "cluster-with-topology-simple-workflow-5q05a5" from the provided yaml
  INFO: Applying the cluster template yaml of cluster cluster-with-topology-simple-workflow-43cwdq/cluster-with-topology-simple-workflow-5q05a5
Running kubectl apply --kubeconfig /var/folders/l3/8dxq0z892n992ffkmb33003r0000gn/T/e2e-kind314256045 -f -
stderr:

stdout:
kubeadmconfigtemplate.bootstrap.cluster.x-k8s.io/nutanix-quick-start-kcfg-0 created
clusterclass.cluster.x-k8s.io/nutanix-quick-start created
kubeadmcontrolplanetemplate.controlplane.cluster.x-k8s.io/nutanix-quick-start-kcpt created
nutanixclustertemplate.infrastructure.cluster.x-k8s.io/nutanix-quick-start-nct created
nutanixmachinetemplate.infrastructure.cluster.x-k8s.io/nutanix-quick-start-cp-nmt created
nutanixmachinetemplate.infrastructure.cluster.x-k8s.io/nutanix-quick-start-md-nmt created
configmap/cluster-with-topology-simple-workflow-5q05a5-pc-trusted-ca-bundle created
configmap/cni-cluster-with-topology-simple-workflow-5q05a5-crs-cni created
configmap/nutanix-ccm created
secret/cluster-with-topology-simple-workflow-5q05a5-pc-creds created
secret/nutanix-ccm-secret created
clusterresourceset.addons.cluster.x-k8s.io/cluster-with-topology-simple-workflow-5q05a5-crs-cni created
clusterresourceset.addons.cluster.x-k8s.io/nutanix-ccm-crs created
cluster.cluster.x-k8s.io/cluster-with-topology-simple-workflow-5q05a5 created

  INFO: Waiting for the cluster infrastructure of cluster cluster-with-topology-simple-workflow-43cwdq/cluster-with-topology-simple-workflow-5q05a5 to be provisioned
  STEP: Waiting for cluster to enter the provisioned phase @ 05/01/24 13:17:49.321
  INFO: Waiting for control plane of cluster cluster-with-topology-simple-workflow-43cwdq/cluster-with-topology-simple-workflow-5q05a5 to be initialized
  INFO: Waiting for the first control plane machine managed by cluster-with-topology-simple-workflow-43cwdq/cluster-with-topology-simple-workflow-5q05a5-6krns to be provisioned
  STEP: Waiting for one control plane node to exist @ 05/01/24 13:17:59.353
  INFO: Waiting for control plane of cluster cluster-with-topology-simple-workflow-43cwdq/cluster-with-topology-simple-workflow-5q05a5 to be ready
  INFO: Waiting for control plane cluster-with-topology-simple-workflow-43cwdq/cluster-with-topology-simple-workflow-5q05a5-6krns to be ready (implies underlying nodes to be ready as well)
  STEP: Waiting for the control plane to be ready @ 05/01/24 13:20:39.737
  STEP: Checking all the control plane machines are in the expected failure domains @ 05/01/24 13:20:39.745
  INFO: Waiting for the machine deployments of cluster cluster-with-topology-simple-workflow-43cwdq/cluster-with-topology-simple-workflow-5q05a5 to be provisioned
  STEP: Waiting for the workload nodes to exist @ 05/01/24 13:20:39.756
  STEP: Checking all the machines controlled by cluster-with-topology-simple-workflow-5q05a5-md-0-tvnnp are in the "<None>" failure domain @ 05/01/24 13:20:39.764
  INFO: Waiting for the machine pools of cluster cluster-with-topology-simple-workflow-43cwdq/cluster-with-topology-simple-workflow-5q05a5 to be provisioned
  STEP: Waiting until nodes are ready @ 05/01/24 13:20:39.778
  STEP: PASSED! @ 05/01/24 13:20:39.967
  STEP: Dumping logs from the "cluster-with-topology-simple-workflow-5q05a5" workload cluster @ 05/01/24 13:20:39.967
Failed to get logs for Machine cluster-with-topology-simple-workflow-5q05a5-6krns-qzbdx, Cluster cluster-with-topology-simple-workflow-43cwdq/cluster-with-topology-simple-workflow-5q05a5: [error creating container exec: Error response from daemon: No such container: cluster-with-topology-simple-workflow-5q05a5-6krns-qzbdx, : error creating container exec: Error response from daemon: No such container: cluster-with-topology-simple-workflow-5q05a5-6krns-qzbdx]
Failed to get logs for Machine cluster-with-topology-simple-workflow-5q05a5-md-0-tvnnp-nfhj2j2, Cluster cluster-with-topology-simple-workflow-43cwdq/cluster-with-topology-simple-workflow-5q05a5: [error creating container exec: Error response from daemon: No such container: cluster-with-topology-simple-workflow-5q05a5-md-0-tvnnp-nfhj2j2, : error creating container exec: Error response from daemon: No such container: cluster-with-topology-simple-workflow-5q05a5-md-0-tvnnp-nfhj2j2]
Failed to get infrastructure logs for Cluster cluster-with-topology-simple-workflow-43cwdq/cluster-with-topology-simple-workflow-5q05a5: failed to inspect container "cluster-with-topology-simple-workflow-5q05a5-lb": Error response from daemon: No such container: cluster-with-topology-simple-workflow-5q05a5-lb
  STEP: Dumping all the Cluster API resources in the "cluster-with-topology-simple-workflow-43cwdq" namespace @ 05/01/24 13:20:40.031
  STEP: Deleting cluster cluster-with-topology-simple-workflow-43cwdq/cluster-with-topology-simple-workflow-5q05a5 @ 05/01/24 13:20:40.301
  STEP: Deleting cluster cluster-with-topology-simple-workflow-43cwdq/cluster-with-topology-simple-workflow-5q05a5 @ 05/01/24 13:20:40.4
  INFO: Waiting for the Cluster cluster-with-topology-simple-workflow-43cwdq/cluster-with-topology-simple-workflow-5q05a5 to be deleted
  STEP: Waiting for cluster cluster-with-topology-simple-workflow-43cwdq/cluster-with-topology-simple-workflow-5q05a5 to be deleted @ 05/01/24 13:20:40.48
  STEP: Deleting namespace used for hosting the "cluster-with-topology-simple-workflow" test spec @ 05/01/24 13:21:00.499
  INFO: Deleting namespace cluster-with-topology-simple-workflow-43cwdq
• [193.134 seconds]
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
[SynchronizedAfterSuite] 
/Users/deepak.muley/go/src/github.com/nutanix-cloud-native/cluster-api-provider-nutanix/test/e2e/e2e_suite_test.go:118
  STEP: Dumping logs from the bootstrap cluster @ 05/01/24 13:21:00.507
  STEP: Tearing down the management cluster @ 05/01/24 13:21:00.879
  INFO: Error getting pod capi-kubeadm-bootstrap-system/capi-kubeadm-bootstrap-controller-manager-55c8bd9999-d5psc, container manager: Get "https://127.0.0.1:53661/api/v1/namespaces/capi-kubeadm-bootstrap-system/pods/capi-kubeadm-bootstrap-controller-manager-55c8bd9999-d5psc": dial tcp 127.0.0.1:53661: connect: connection refused - error from a previous attempt: EOF
  INFO: Error getting pod capi-system/capi-controller-manager-748d4cf778-f2lj5, container manager: Get "https://127.0.0.1:53661/api/v1/namespaces/capi-system/pods/capi-controller-manager-748d4cf778-f2lj5": dial tcp 127.0.0.1:53661: connect: connection refused - error from a previous attempt: EOF
  INFO: Error getting pod capx-system/capx-controller-manager-76647446-jsrzz, container kube-rbac-proxy: Get "https://127.0.0.1:53661/api/v1/namespaces/capx-system/pods/capx-controller-manager-76647446-jsrzz": dial tcp 127.0.0.1:53661: connect: connection refused - error from a previous attempt: EOF
  INFO: Error getting pod capi-kubeadm-control-plane-system/capi-kubeadm-control-plane-controller-manager-8458749949-ztdwl, container manager: Get "https://127.0.0.1:53661/api/v1/namespaces/capi-kubeadm-control-plane-system/pods/capi-kubeadm-control-plane-controller-manager-8458749949-ztdwl": dial tcp 127.0.0.1:53661: connect: connection refused - error from a previous attempt: EOF
  INFO: Error getting pod capx-system/capx-controller-manager-76647446-jsrzz, container manager: Get "https://127.0.0.1:53661/api/v1/namespaces/capx-system/pods/capx-controller-manager-76647446-jsrzz": dial tcp 127.0.0.1:53661: connect: connection refused - error from a previous attempt: EOF
[SynchronizedAfterSuite] PASSED [2.260 seconds]
------------------------------
[ReportAfterSuite] Autogenerated ReportAfterSuite for --junit-report
autogenerated by Ginkgo
[ReportAfterSuite] PASSED [0.011 seconds]
------------------------------

Ran 1 of 97 Specs in 335.521 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 96 Skipped
PASS
</pre>

_Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and test output_

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by deepakm-ntnx** [comment]
fyi
<pre>
RUN_VALIDATION_TESTS_ONLY=true make test-e2e-calico LABEL_FILTERS="Kube130 && cluster-topology-upgrade"
</pre>

<pre>
When upgrading the k8s version of cluster with topology Upgrade a cluster with topology from version Kube129 to Kube130 [clusterclass, only-for-validation, cluster-topology-upgrade, Kube129, Kube130]
/Users/deepak.muley/go/src/github.com/nutanix-cloud-native/cluster-api-provider-nutanix/test/e2e/cluster_topology_upgrade_test.go:139
  STEP: Creating a namespace for hosting the "cluster-with-topology-upgrade-workflow" test spec @ 05/01/24 14:07:48.077
  INFO: Creating namespace cluster-with-topology-upgrade-workflow-niygr3
  INFO: Creating event watcher for namespace "cluster-with-topology-upgrade-workflow-niygr3"
  STEP: Creating a workload cluster with topology @ 05/01/24 14:07:48.089
  INFO: Creating the workload cluster with name "cluster-with-topology-upgrade-workflow-tv96rc" using the "topology" template (Kubernetes v1.29.4, 1 control-plane machines, 1 worker machines)
  INFO: Getting the cluster template yaml
  INFO: clusterctl config cluster cluster-with-topology-upgrade-workflow-tv96rc --infrastructure (default) --kubernetes-version v1.29.4 --control-plane-machine-count 1 --worker-machine-count 1 --flavor topology
  INFO: Creating the workload cluster with name "cluster-with-topology-upgrade-workflow-tv96rc" from the provided yaml
  INFO: Applying the cluster template yaml of cluster cluster-with-topology-upgrade-workflow-niygr3/cluster-with-topology-upgrade-workflow-tv96rc
Running kubectl apply --kubeconfig /var/folders/l3/8dxq0z892n992ffkmb33003r0000gn/T/e2e-kind2862970233 -f -
stderr:

stdout:
kubeadmconfigtemplate.bootstrap.cluster.x-k8s.io/nutanix-quick-start-kcfg-0 created
clusterclass.cluster.x-k8s.io/nutanix-quick-start created
kubeadmcontrolplanetemplate.controlplane.cluster.x-k8s.io/nutanix-quick-start-kcpt created
nutanixclustertemplate.infrastructure.cluster.x-k8s.io/nutanix-quick-start-nct created
nutanixmachinetemplate.infrastructure.cluster.x-k8s.io/nutanix-quick-start-cp-nmt created
nutanixmachinetemplate.infrastructure.cluster.x-k8s.io/nutanix-quick-start-md-nmt created
configmap/cluster-with-topology-upgrade-workflow-tv96rc-pc-trusted-ca-bundle created
configmap/cni-cluster-with-topology-upgrade-workflow-tv96rc-crs-cni created
configmap/nutanix-ccm created
secret/cluster-with-topology-upgrade-workflow-tv96rc-pc-creds created
secret/nutanix-ccm-secret created
clusterresourceset.addons.cluster.x-k8s.io/cluster-with-topology-upgrade-workflow-tv96rc-crs-cni created
clusterresourceset.addons.cluster.x-k8s.io/nutanix-ccm-crs created
cluster.cluster.x-k8s.io/cluster-with-topology-upgrade-workflow-tv96rc created

  INFO: Waiting for the cluster infrastructure of cluster cluster-with-topology-upgrade-workflow-niygr3/cluster-with-topology-upgrade-workflow-tv96rc to be provisioned
  STEP: Waiting for cluster to enter the provisioned phase @ 05/01/24 14:07:50.234
  INFO: Waiting for control plane of cluster cluster-with-topology-upgrade-workflow-niygr3/cluster-with-topology-upgrade-workflow-tv96rc to be initialized
  INFO: Waiting for the first control plane machine managed by cluster-with-topology-upgrade-workflow-niygr3/cluster-with-topology-upgrade-workflow-tv96rc-8zbw4 to be provisioned
  STEP: Waiting for one control plane node to exist @ 05/01/24 14:08:00.264
  INFO: Waiting for control plane of cluster cluster-with-topology-upgrade-workflow-niygr3/cluster-with-topology-upgrade-workflow-tv96rc to be ready
  INFO: Waiting for control plane cluster-with-topology-upgrade-workflow-niygr3/cluster-with-topology-upgrade-workflow-tv96rc-8zbw4 to be ready (implies underlying nodes to be ready as well)
  STEP: Waiting for the control plane to be ready @ 05/01/24 14:09:30.333
  STEP: Checking all the control plane machines are in the expected failure domains @ 05/01/24 14:09:30.343
  INFO: Waiting for the machine deployments of cluster cluster-with-topology-upgrade-workflow-niygr3/cluster-with-topology-upgrade-workflow-tv96rc to be provisioned
  STEP: Waiting for the workload nodes to exist @ 05/01/24 14:09:30.361
  STEP: Checking all the machines controlled by cluster-with-topology-upgrade-workflow-tv96rc-md-0-zjpvn are in the "<None>" failure domain @ 05/01/24 14:09:30.372
  INFO: Waiting for the machine pools of cluster cluster-with-topology-upgrade-workflow-niygr3/cluster-with-topology-upgrade-workflow-tv96rc to be provisioned
  STEP: Waiting until nodes are ready @ 05/01/24 14:09:30.39
  INFO: Creating the workload cluster with name "cluster-with-topology-upgrade-workflow-tv96rc" using the "topology" template (Kubernetes v1.30.0, 1 control-plane machines, 1 worker machines)
  INFO: Getting the cluster template yaml
  INFO: clusterctl config cluster cluster-with-topology-upgrade-workflow-tv96rc --infrastructure (default) --kubernetes-version v1.30.0 --control-plane-machine-count 1 --worker-machine-count 1 --flavor topology
  INFO: Creating the workload cluster with name "cluster-with-topology-upgrade-workflow-tv96rc" from the provided yaml
  INFO: Applying the cluster template yaml of cluster cluster-with-topology-upgrade-workflow-niygr3/cluster-with-topology-upgrade-workflow-tv96rc
Running kubectl apply --kubeconfig /var/folders/l3/8dxq0z892n992ffkmb33003r0000gn/T/e2e-kind2862970233 -f -
stderr:

stdout:
configmap/cluster-with-topology-upgrade-workflow-tv96rc-pc-trusted-ca-bundle unchanged
configmap/cni-cluster-with-topology-upgrade-workflow-tv96rc-crs-cni unchanged
configmap/nutanix-ccm unchanged
secret/cluster-with-topology-upgrade-workflow-tv96rc-pc-creds configured
secret/nutanix-ccm-secret configured
clusterresourceset.addons.cluster.x-k8s.io/cluster-with-topology-upgrade-workflow-tv96rc-crs-cni unchanged
clusterresourceset.addons.cluster.x-k8s.io/nutanix-ccm-crs unchanged
cluster.cluster.x-k8s.io/cluster-with-topology-upgrade-workflow-tv96rc configured

  INFO: Waiting for the cluster infrastructure of cluster cluster-with-topology-upgrade-workflow-niygr3/cluster-with-topology-upgrade-workflow-tv96rc to be provisioned
  STEP: Waiting for cluster to enter the provisioned phase @ 05/01/24 14:10:02.867
  INFO: Waiting for control plane of cluster cluster-with-topology-upgrade-workflow-niygr3/cluster-with-topology-upgrade-workflow-tv96rc to be initialized
  INFO: Waiting for the first control plane machine managed by cluster-with-topology-upgrade-workflow-niygr3/cluster-with-topology-upgrade-workflow-tv96rc-8zbw4 to be provisioned
  STEP: Waiting for one control plane node to exist @ 05/01/24 14:10:02.9
  INFO: Waiting for control plane of cluster cluster-with-topology-upgrade-workflow-niygr3/cluster-with-topology-upgrade-workflow-tv96rc to be ready
  INFO: Waiting for control plane cluster-with-topology-upgrade-workflow-niygr3/cluster-with-topology-upgrade-workflow-tv96rc-8zbw4 to be ready (implies underlying nodes to be ready as well)
  STEP: Waiting for the control plane to be ready @ 05/01/24 14:10:02.91
  STEP: Checking all the control plane machines are in the expected failure domains @ 05/01/24 14:10:02.919
  INFO: Waiting for the machine deployments of cluster cluster-with-topology-upgrade-workflow-niygr3/cluster-with-topology-upgrade-workflow-tv96rc to be provisioned
  STEP: Waiting for the workload nodes to exist @ 05/01/24 14:10:02.941
  STEP: Checking all the machines controlled by cluster-with-topology-upgrade-workflow-tv96rc-md-0-zjpvn are in the "<None>" failure domain @ 05/01/24 14:10:02.953
  INFO: Waiting for the machine pools of cluster cluster-with-topology-upgrade-workflow-niygr3/cluster-with-topology-upgrade-workflow-tv96rc to be provisioned
  STEP: Waiting for control-plane machines to have the upgraded Kubernetes version @ 05/01/24 14:10:02.98
  STEP: Ensuring all control-plane machines have upgraded kubernetes version v1.30.0 @ 05/01/24 14:10:02.981
  STEP: Waiting for machine deployment machines to have the upgraded Kubernetes version @ 05/01/24 14:12:33.156
  INFO: Ensuring all MachineDeployment Machines have upgraded kubernetes version v1.30.0
  STEP: Waiting until nodes are ready @ 05/01/24 14:13:13.208
  STEP: PASSED! @ 05/01/24 14:13:13.368
  STEP: Dumping logs from the "cluster-with-topology-upgrade-workflow-tv96rc" workload cluster @ 05/01/24 14:13:13.368
Failed to get logs for Machine cluster-with-topology-upgrade-workflow-tv96rc-8zbw4-xwxw7, Cluster cluster-with-topology-upgrade-workflow-niygr3/cluster-with-topology-upgrade-workflow-tv96rc: [error creating container exec: Error response from daemon: No such container: cluster-with-topology-upgrade-workflow-tv96rc-8zbw4-xwxw7, : error creating container exec: Error response from daemon: No such container: cluster-with-topology-upgrade-workflow-tv96rc-8zbw4-xwxw7]
Failed to get logs for Machine cluster-with-topology-upgrade-workflow-tv96rc-md-0-zjpvn-dvkcp2, Cluster cluster-with-topology-upgrade-workflow-niygr3/cluster-with-topology-upgrade-workflow-tv96rc: [error creating container exec: Error response from daemon: No such container: cluster-with-topology-upgrade-workflow-tv96rc-md-0-zjpvn-dvkcp2, : error creating container exec: Error response from daemon: No such container: cluster-with-topology-upgrade-workflow-tv96rc-md-0-zjpvn-dvkcp2]
Failed to get infrastructure logs for Cluster cluster-with-topology-upgrade-workflow-niygr3/cluster-with-topology-upgrade-workflow-tv96rc: failed to inspect container "cluster-with-topology-upgrade-workflow-tv96rc-lb": Error response from daemon: No such container: cluster-with-topology-upgrade-workflow-tv96rc-lb
  STEP: Dumping all the Cluster API resources in the "cluster-with-topology-upgrade-workflow-niygr3" namespace @ 05/01/24 14:13:13.552
  STEP: Deleting cluster cluster-with-topology-upgrade-workflow-niygr3/cluster-with-topology-upgrade-workflow-tv96rc @ 05/01/24 14:13:14.152
  STEP: Deleting cluster cluster-with-topology-upgrade-workflow-niygr3/cluster-with-topology-upgrade-workflow-tv96rc @ 05/01/24 14:13:14.19
  INFO: Waiting for the Cluster cluster-with-topology-upgrade-workflow-niygr3/cluster-with-topology-upgrade-workflow-tv96rc to be deleted
  STEP: Waiting for cluster cluster-with-topology-upgrade-workflow-niygr3/cluster-with-topology-upgrade-workflow-tv96rc to be deleted @ 05/01/24 14:13:14.214
  STEP: Deleting namespace used for hosting the "cluster-with-topology-upgrade-workflow" test spec @ 05/01/24 14:13:44.277
  INFO: Deleting namespace cluster-with-topology-upgrade-workflow-niygr3
• [356.203 seconds]
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
[SynchronizedAfterSuite] 
/Users/deepak.muley/go/src/github.com/nutanix-cloud-native/cluster-api-provider-nutanix/test/e2e/e2e_suite_test.go:118
  STEP: Dumping logs from the bootstrap cluster @ 05/01/24 14:13:44.288
  STEP: Tearing down the management cluster @ 05/01/24 14:13:44.886
  INFO: Error getting pod capx-system/capx-controller-manager-76647446-vns42, container kube-rbac-proxy: Get "https://127.0.0.1:53919/api/v1/namespaces/capx-system/pods/capx-controller-manager-76647446-vns42": dial tcp 127.0.0.1:53919: connect: connection refused - error from a previous attempt: EOF
  INFO: Error getting pod capi-system/capi-controller-manager-748d4cf778-mrx5j, container manager: Get "https://127.0.0.1:53919/api/v1/namespaces/capi-system/pods/capi-controller-manager-748d4cf778-mrx5j": dial tcp 127.0.0.1:53919: connect: connection refused - error from a previous attempt: EOF
  INFO: Error getting pod capi-kubeadm-control-plane-system/capi-kubeadm-control-plane-controller-manager-8458749949-dk544, container manager: Get "https://127.0.0.1:53919/api/v1/namespaces/capi-kubeadm-control-plane-system/pods/capi-kubeadm-control-plane-controller-manager-8458749949-dk544": dial tcp 127.0.0.1:53919: connect: connection refused - error from a previous attempt: EOF
  INFO: Error getting pod capx-system/capx-controller-manager-76647446-vns42, container manager: Get "https://127.0.0.1:53919/api/v1/namespaces/capx-system/pods/capx-controller-manager-76647446-vns42": dial tcp 127.0.0.1:53919: connect: connection refused - error from a previous attempt: EOF
  INFO: Error getting pod capi-kubeadm-bootstrap-system/capi-kubeadm-bootstrap-controller-manager-55c8bd9999-xx7lk, container manager: Get "https://127.0.0.1:53919/api/v1/namespaces/capi-kubeadm-bootstrap-system/pods/capi-kubeadm-bootstrap-controller-manager-55c8bd9999-xx7lk": dial tcp 127.0.0.1:53919: connect: connection refused - error from a previous attempt: EOF
[SynchronizedAfterSuite] PASSED [3.150 seconds]
------------------------------
[ReportAfterSuite] Autogenerated ReportAfterSuite for --junit-report
autogenerated by Ginkgo
[ReportAfterSuite] PASSED [0.009 seconds]
------------------------------

Ran 1 of 97 Specs in 503.258 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 96 Skipped
PASS
</pre>

File: test/e2e/config/nutanix.yaml [hunk 1]
```diff
@@ -233,26 +233,26 @@ variables:
...
+  NUTANIX_MACHINE_TEMPLATE_IMAGE_NAME_v1_27: "ubuntu-2204-kube-v1.27.13.qcow2"
+  NUTANIX_MACHINE_TEMPLATE_IMAGE_NAME_v1_28: "ubuntu-2204-kube-v1.28.9.qcow2"
+  NUTANIX_MACHINE_TEMPLATE_IMAGE_NAME_v1_29: "ubuntu-2204-kube-v1.29.4.qcow2"
+  NUTANIX_MACHINE_TEMPLATE_IMAGE_NAME_v1_30: "ubuntu-2204-kube-v1.30.0.qcow2"
   CONTROL_PLANE_MACHINE_TEMPLATE_UPGRADE_TO: "cluster-upgrade-conformance"
   WORKERS_MACHINE_TEMPLATE_UPGRADE_TO: "cluster-upgrade-conformance"
-  NUTANIX_MACHINE_TEMPLATE_IMAGE_UPGRADE_TO: "ubuntu-2204-kube-v1.29.2.qcow2"
-  NUTANIX_MACHINE_TEMPLATE_IMAGE_UPGRADE_FROM: "ubuntu-2204-kube-v1.28.7.qcow2"
+  NUTANIX_MACHINE_TEMPLATE_IMAGE_UPGRADE_TO: "ubuntu-2204-kube-v1.30.0.qcow2"
+  NUTANIX_MACHINE_TEMPLATE_IMAGE_UPGRADE_FROM: "ubuntu-2204-kube-v1.29.4.qcow2"
```
Comment (thunderboltsid) [inline test/e2e/config/nutanix.yaml:255]: It'd be good to think of some automation to keep this list up to date by creating PRs. similar to dependabot but for images.

Comment (deepakm-ntnx) [reply test/e2e/config/nutanix.yaml:255]: we can do that when we get them in CI runs. currently they all run manually

**Review by adiantum — APPROVED** [review]
/lgtm
/approve

File: test/e2e/cluster_topology.go [hunk 2]
```diff
@@ -31,6 +31,51 @@ import (
...
+	targetKube   *E2eConfigK8SVersion
+	targetLabels []string
+}
+
+var SupportedK8STargetConfigs = []TestConfig{
+	{targetKube: NewE2eConfigK8SVersion127(), targetLabels: []string{"Kube127"}},
+	{targetKube: NewE2eConfigK8SVersion128(), targetLabels: []string{"Kube128"}},
+	{targetKube: NewE2eConfigK8SVersion129(), targetLabels: []string{"Kube129"}},
+	{targetKube: NewE2eConfigK8SVersion130(), targetLabels: []string{"Kube130"}},
+}
```
Comment (thunderboltsid) [inline test/e2e/cluster_topology.go:77]: please add comments for each exported type

**Review by thunderboltsid — APPROVED** [review]

**Pull request merged**